app/services/users/delete.rb

```diff
@@ -10,6 +10,7 @@ def call
       delete_user_activity
       user.unsubscribe_from_newsletters
       EdgeCache::Bust.call("/#{user.username}")
+      Users::Suspended.create_from_user(user) if user.has_role?(:banned)
```

Comment:
Might want to add this `if` inside `create_from_user` and plan ahead for the `suspended` role that's coming down the pipeline in #12270

Reply:
I had considered this, but it seemed confusing for readers of this service, as if we're unconditionally doing this for everyone. How about putting the check in a helper method in the same service?

Reply:
I think it can stay as it is, what about adding `|| user.has_role?(:suspended)` too? Mostly so that if this is merged ahead of #12270 it won't slip from the check

Reply:
Added.